### PR TITLE
fix: optimizely audiences name

### PR DESCRIPTION
**Pull request by maheshkutty**
## PR Description

While building the audience name, an error occurred, which caused the integration to fail during initialization.
In this PR, we added a separate function to safely fetch the audience name

## Linear task (optional)

[Resolves INT-5858](https://linear.app/rudderstack/issue/INT-5858/handle-null-audience-names-from-optimizely-in-integration)

## Cross Browser Tests

Please confirm you have tested for the following browsers:

- [ ] Chrome
- [ ] Firefox
- [ ] IE11

## Sanity Suite

- [x] All sanity suite test cases pass locally

## Security

- [ ] The code changed/added as part of this pull request won't create any security issues with how the software is being used.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved error handling and resilience for Optimizely integration state management.
  * Enhanced audience data processing with better handling of edge cases.

* **Tests**
  * Added comprehensive test coverage for Optimizely browser integration.

### CODE REVIEW

File: packages/analytics-js-integrations/src/integrations/Optimizely/browser.js [hunk 1]
```diff
@@ -123,7 +134,10 @@ class Optimizely {
           sendCampaignData(campaignState);
         }
       } catch (e) {
-        logger.error('Page loaded without Optimizely.');
+        logger.error(
+          "Optimizely not initialized due to transformation error: ",
+          e.message,
+        );
```
Comment (saikumarrs) [inline packages/analytics-js-integrations/src/integrations/Optimizely/browser.js]: The message can be more contextual.
```suggestion
        logger.error(
          "Couldn't successfully process active campaign data from Optimizely",
          e,
        );
```

File: packages/analytics-js-integrations/src/integrations/Optimizely/browser.js
Comment (saikumarrs) [inline packages/analytics-js-integrations/src/integrations/Optimizely/browser.js:149]: We should properly error handle this function too.

**Review by saikumarrs — CHANGES_REQUESTED** [review]

File: packages/analytics-js-integrations/src/integrations/Optimizely/browser.js (hunk 1 above)
Comment (maheshkutty) [reply packages/analytics-js-integrations/src/integrations/Optimizely/browser.js]: Done

**Review by saikumarrs — DISMISSED** [review]

File: packages/analytics-js-integrations/src/integrations/Optimizely/browser.js
Comment (maheshkutty) [reply packages/analytics-js-integrations/src/integrations/Optimizely/browser.js:149]: created a separate function to getState

**Review by saikumarrs — DISMISSED** [review]

**Review by saikumarrs — APPROVED** [review]
